test/runtests.jl

```diff
@@ -68,6 +68,120 @@ end
+    df[:label_1] = round.(Int, rand(num_rows))
+
+    Random.seed!(9999)
+    OLS_1 = GLM.lm(
+        StatsModels.@formula(
+            label_1 ~ 1 + feature_1 + feature_2 + feature_3
+            ),
+        df,
+        true,
+        )
+    Test.@test(0.4 < GLM.coeftable(OLS_1.model).cols[1][1] < 0.6)
+    Test.@test(isapprox(GLM.coeftable(OLS_1.model).cols[1][2], 0.0))
+    Test.@test(0.003 < GLM.coeftable(OLS_1.model).cols[1][3] < 0.004)
+    Test.@test(0.004 < GLM.coeftable(OLS_1.model).cols[1][4] < 0.005)
+    Test.@test(GLM.coeftable(OLS_1.model).cols[4][1].v < 1e-10)
+    Test.@test(isnan(GLM.coeftable(OLS_1.model).cols[4][2].v))
+    Test.@test(0.8 < GLM.coeftable(OLS_1.model).cols[4][3].v < 1.0)
+    Test.@test(0.9 < GLM.coeftable(OLS_1.model).cols[4][4].v < 1.0)
+
+    Random.seed!(9999)
+    Test.@test_throws(
+        LinearAlgebra.PosDefException,
+        GLM.glm(
+            StatsModels.@formula(
+                label_1 ~ 1 + feature_1 + feature_2 + feature_3
+                ),
+            df,
+            GLM.Binomial();
+            allowrankdeficient = false,
+            ),
+        )
+
+    Random.seed!(9999)
+    LR_1 = GLM.glm(
+        StatsModels.@formula(
+            label_1 ~ 1 + feature_1 + feature_2 + feature_3
+            ),
+        df, GLM.Binomial();
+        allowrankdeficient = true,
+        )
+    Test.@test(-0.1 < GLM.coeftable(LR_1.model).cols[1][1] < 0.0)
+    Test.@test(GLM.coeftable(LR_1.model).cols[1][2] < -4e12)
+    Test.@test(GLM.coeftable(LR_1.model).cols[1][3] > 2e12)
+    Test.@test(0.01 < GLM.coeftable(LR_1.model).cols[1][4] < 0.03)
+    Test.@test(0.6 < GLM.coeftable(LR_1.model).cols[4][1].v < 0.8)
+    Test.@test(isnan(GLM.coeftable(LR_1.model).cols[4][2].v))
+    Test.@test(GLM.coeftable(LR_1.model).cols[4][3].v < 1e-90)
+    Test.@test(0.9 < GLM.coeftable(LR_1.model).cols[4][4].v < 1.0)
+
+    Random.seed!(123)
+    df[:label_2] = Vector{Int}(undef, num_rows)
+    for i = 1:num_rows
+        if df[i, :feature_3] > 0
+            df[i, :label_2] = Int( rand() > 0.2)
+        else
+            df[i, :label_2] = Int( rand() > 0.8)
+        end
+    end
+    Random.seed!(123)
+    OLS_2 = GLM.lm(
+        StatsModels.@formula(
+            label_2 ~ 1 + feature_1 + feature_2 + feature_3
+            ),
+        df,
+        true,
+        )
+    Test.@test(0.5 < GLM.coeftable(OLS_2.model).cols[1][1] < 0.6)
+    Test.@test(isapprox(GLM.coeftable(OLS_2.model).cols[1][2], 0.0))
+    Test.@test(-0.003 > GLM.coeftable(OLS_2.model).cols[1][3] > -0.005)
+    Test.@test(0.2 < GLM.coeftable(OLS_2.model).cols[1][4] < 0.3)
+    Test.@test(GLM.coeftable(OLS_2.model).cols[4][1].v < 1e-10)
+    Test.@test(isnan(GLM.coeftable(OLS_2.model).cols[4][2].v))
+    Test.@test(0.8 < GLM.coeftable(OLS_2.model).cols[4][3].v < 1.0)
+    Test.@test(GLM.coeftable(OLS_2.model).cols[4][4].v < 1e-5)
+
+    Random.seed!(123)
+    Test.@test_throws(
+        LinearAlgebra.PosDefException,
```

Comment:
Same comments as above and make the test sensible to default options (not specifying it to catch any errors if changes to these)
```
@test_throws(PosDefException,
             glm(@formula(label_2 ~ 1 + feature_1 + feature_2 + feature_3),
                 df,
                 Binomial()))
```